src/core/state.go

```diff
@@ -216,6 +216,52 @@ type BuildState struct {
 	// EnableBreakpoints enablese the breakpoint() build-in, and drops Please into an interactive debugger when
 	// they're encountered.
 	EnableBreakpoints bool
+
+	// initOnce is used to control when we load plugin configuration, and trigger parser initialisation. We need access
+	// to the subrepo's config files to do this so this is deferred to build time.
+	initOnce *sync.Once
+	// Used to synchronise with the above once to make sure we've loaded the subrepo config before trying to set up the
+	// CONFIG.PLUGIN_NAME during subincludes
+	initialised chan struct{}
+}
+
+// Copy creates a copy of this state object
+func (state *BuildState) Copy() *BuildState {
+	ret := &BuildState{}
+	*ret = *state
+
+	ret.initialised = make(chan struct{})
+	ret.initOnce = new(sync.Once)
+	return ret
+}
+
+// Initialise will load the .plzconfig from the subrepo. We can only do this once the subrepo is built hence why
+// it's not done up front. Once we have done that, we can initialise the parser for the subrepo.
+func (state *BuildState) Initialise(subrepo *Subrepo) (err error) {
+	state.initOnce.Do(func() {
+		defer close(state.initialised)
+
+		// If we are the root repo, or an cross-compilation of that, we don't want to re-load the config files. That's
+		// handled for us already in plz.go
+		if state.CurrentSubrepo != "" {
+			state.RepoConfig = &Configuration{}
+			err = readConfigFilesInto(state.RepoConfig, append(subrepo.AdditionalConfigFiles, filepath.Join(subrepo.Root, ".plzconfig")))
+			if err != nil {
+				return
+			}
+			if err = validateSubrepoNameAndPluginConfig(state.Config, state.RepoConfig, subrepo); err != nil {
+				return
+			}
+		}
+
+		go state.Parser.Init(state)
+	})
+	return
```

Comment:
Do we need to return nil or error here depending on how `Do` went?

Reply:
Ah, this is a go thing. We name the return variable, so when we can set it in the closure. When we return like this, we return that variable from the function. It's a bit weird but useful in this sort of situation.

Reply:
Oh I see, ok that makes sense then. Thanks!